pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -83,16 +92,18 @@ type stateMachineMsg struct {
 type Service struct {
 	service.Action
 	service.Message
-	ctx             *context
-	callbackChannel chan *message
-	connectionStore *connectionStore
+	ctx                *context
+	callbackChannel    chan *message
+	connectionRecorder *connection.Recorder
+	connectionStore    didstore.ConnectionStore
```

Comment:
since a new `connectionRecoder` is created, I would move `connectionStore` into it.

`Service` doesn't need to handle 2 struct types for connection, 1 is enough.

I would also make this call directly from `connectionRecoder` as well: https://github.com/hyperledger/aries-framework-go/pull/2588/files#diff-a4b94fca66b9a726621a9536a1083e4c60cc89a64ad65094608a7c42a25ca702R333


this way the service only handles 1 struct for connections only

Reply:
this is exactly what we had before and what I was trying to avoid - problem with having a single struct for these different things is that it becomes confused which one is used - and they are very different: connectionRecorder is "lightweight" transient storage and connectionStore is expensive one that can be backed by heavy thing like EDV

Reply:
but the service doesn't need to be aware of the what/where the connectionStore lies. The connectionRecorder handles this struct on behalf of the service and can easily change it without the service's knowledge. This separation of concern helps keep the service cleaner and requires less maintenance.

Reply:
another option would be to create a `connectionManager` that has a `connectionRecorder` and `connectionStore` and have the service include a `connectionManager` instead

Reply:
I have mixed feelings about that. With the previous structure
```
type connectionStore struct {
	*connection.Recorder
	*did.ConnectionStore
}
```
it was not immediately clear whether `c.SaveConnectionRecord` belongs to "cheap" `connection.Recorder` or to "expensive" `did.ConnectionStore`. It makes sense for that sort of information to be explicit. In my opinion, this is what service should NOT hide.

Reply:
Guys, do you have any thoughts here?^

Reply:
that's fine to keep it as is, I'm ok with this for now.